File: P02 - Kelas Objek/Java/Bag. kedua/Buah.java
package Praktikum.P02;

public class Buah {
    public String nama;
    public String warna;
    public String rasa;
    public int berat;

    public Buah(String nama, String warna, String rasa, int berat) {
        this.nama = nama;
        this.warna = warna;
        this.rasa = rasa;
        this.berat = berat;

        if (this.berat < 1) {
            this.berat = 1;
        }
    }

    public int hitung_harga() {
        return (this.berat * 5000);
    }

    public void tampilkan() {
        System.out.println("Nama  : " + this.nama);
        System.out.println("Warna : " + this.warna);
        System.out.println("Rasa  : " + this.rasa);
        System.out.println("Berat : " + this.berat + " kg");
        System.out.print("Harga : " + this.hitung_harga());
        System.out.println("\n");
    }

    public void ubahRasa(String rasaBaru) {
        this.rasa = rasaBaru;
    }

    public int hitung_harga_diskon() {
        if (this.berat > 5) {
            return (int) (this.hitung_harga() * 0.9); // Diskon 10% jika berat lebih dari 5 kg
        } else {
            return this.hitung_harga();
        }
    }

    public static void main(String[] args) {
        Buah b1 = new Buah("Apel", "Merah", "Manis", 2);
        Buah b2 = new Buah("Alpukat", "Hijau", "Manis", 2);
        Buah b3 = new Buah("Jeruk", "Kuning", "Manis Asam", 3);

        System.out.println("Macam-macam Buah yang dibeli: ");
        System.out.println("Buah 1.");
        b1.tampilkan();
        System.out.println("Buah 2.");
        b2.tampilkan();
        System.out.println("Buah 3.");
        b3.tampilkan();

        b2.ubahRasa("Asam Manis");
        System.out.println("Rasa Buah 2 setelah diubah: " + b2.rasa);
        System.out.println("Harga Buah 3 setelah diskon: " + b3.hitung_harga_diskon());
    }
}

File: P02 - Kelas Objek/Python/Bag. kedua/Buah.py
# File: Buah.py
# Penulis : Elmosius Suli
# Membuat kelas Buah

class Buah:
    def __init__(self, nama:str, warna:str, rasa:str, berat:int) -> None:
        self.nama = nama 
        self.warna = warna
        self.rasa = rasa
        self.berat = berat
        # kalau beratnya kurang dari 1 kg
        if (self.berat < 1):
            self.berat = 1
        return
    
    def hitung_harga(self) -> int :
        return (self.berat*5000)
    
    def tampilkan(self) -> None:
        print("Nama  : ", self.nama, "\nWarna : ", self.warna, "\nRasa  : ", self.rasa, "\nBerat : ", self.berat, "kg")
        print("Harga : ", self.hitung_harga(), '\n')
        return
    
## Program utama ##
def main():
    b1 = Buah("Semangka", "Hijau", "Manis", 3)
    b2 = Buah("Pisang", "Kuning", "Manis", 2)
    b3 = Buah("Jeruk", "Kuning", "Manis Asam", 0.2)

    print("Macam-macam Buah: \n")
    print("Buah 1. ")
    b1.tampilkan()
    
    print("Buah 2. ")
    b2.tampilkan()
    
    print("Buah 3. ")
    b3.tampilkan()

    print("="*40)

    harga = 0
    berat = 0
    pilih = int(input("Kode buah yang ingin dibeli (0 untuk selesai) : "))
    while (pilih != 0):
        if (pilih == 1):
            harga += b1.hitung_harga()
            berat += b1.berat
            print(f"Membeli {b1.nama} seberat {b1.berat} kg seharga Rp{b1.hitung_harga()},-")
        elif (pilih == 2):
            harga += b2.hitung_harga()
            berat += b2.berat
            print(f"Membeli {b2.nama} seberat {b2.berat} kg seharga Rp{b2.hitung_harga()},-")
        elif (pilih == 3):
            harga += b3.hitung_harga()
            berat += b3.berat
            print(f"Membeli {b3.nama} seberat {b3.berat} kg seharga Rp{b3.hitung_harga()},-")
        pilih = int(input("Kode buah yang ingin dibeli (0 untuk selesai) : "))

    print(f"Total berat = {berat} kg")
    print(f"Total belanjaan = Rp{harga},-")
    bayar = int(input("Uang yang dibayar: "))
    print(f"Kembaliannya Rp{bayar-harga},-")
  
if __name__ == '__main__':
    main()
  